tests/test_async_job.py

```diff
@@ -0,0 +1,104 @@
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import asyncio
+import os
+import time
+
+from mock import AsyncMock, patch
+import pytest
+
+from octobot_commons.async_job import AsyncJob
+from octobot_commons.asyncio_tools import wait_asyncio_next_cycle
+
+pytestmark = pytest.mark.asyncio
+
+
+async def callback():
+    pass
+
+
+async def test_should_run():
+    job = AsyncJob(callback)
+    if not os.getenv('CYTHON_IGNORE'):
+        assert job._has_enough_time_elapsed()
+        job.last_execution_time = time.time()
+        assert job._has_enough_time_elapsed()
+    job.stop()
+
+
+async def test_should_run_with_delays():
+    job = AsyncJob(callback, min_execution_delay=1)
+    if not os.getenv('CYTHON_IGNORE'):
+        assert job._has_enough_time_elapsed()
+        job.last_execution_time = time.time()
+        assert not job._has_enough_time_elapsed()
+        await asyncio.sleep(1)
+        assert job._has_enough_time_elapsed()
+    job.stop()
+
+
+async def test_should_run_with_dependencies():
+    job2 = AsyncJob(callback)
+    job3 = AsyncJob(callback)
+    job = AsyncJob(callback)
+    job.add_job_dependency(job2)
+    job.add_job_dependency(job3)
+    if not os.getenv('CYTHON_IGNORE'):
+        assert job._should_run_job()
+        job2.running_task_event.clear()
+        assert not job._should_run_job()
+        job2.running_task_event.clear()
+        job3.running_task_event.clear()
+        assert not job._should_run_job()
+        job2.running_task_event.set()
+        assert not job._should_run_job()
+        job3.running_task_event.set()
+        assert job._should_run_job()
+    job.stop()
+
+
+async def test_clear():
+    AsyncJob(callback).clear()
+
+
+async def test_run():
+    job = AsyncJob(callback, execution_interval_delay=5, min_execution_delay=2)
+    if not os.getenv('CYTHON_IGNORE'):
+        with patch.object(job, 'callback', new=AsyncMock()) as mocked_test_job_callback:
+            await wait_asyncio_next_cycle()
+            mocked_test_job_callback.assert_not_called()
+            assert not job.is_started
+            await job.run()
+            await wait_asyncio_next_cycle()
+            mocked_test_job_callback.assert_called_once()
+            assert job.is_started
+
+            # delay has not been waited
+            await job.run()
+            await wait_asyncio_next_cycle()
+            mocked_test_job_callback.assert_called_once()
+            assert job.is_started
+
+            await asyncio.sleep(3)
+            mocked_test_job_callback.assert_called_once()
+
+            await asyncio.sleep(3)
+            assert mocked_test_job_callback.call_count == 2
+
+            await job.run(force=True, wait_for_task_execution=True)
+            assert mocked_test_job_callback.call_count == 3
```

Comment:
I think it would be great to add these lines:
```python
await asyncio.sleep(1)
# no periodic trigger yet
assert mocked_test_job_callback.call_count == 3

await job.run(force=True, wait_for_task_execution=False)
# task not yet executed
assert mocked_test_job_callback.call_count == 3
await wait_asyncio_next_cycle()
assert mocked_test_job_callback.call_count == 4

await asyncio.sleep(1)
# no periodic auto trigger yet (delay is not yet over)
assert mocked_test_job_callback.call_count == 4

await asyncio.sleep(5)
# periodic auto trigger
assert mocked_test_job_callback.call_count == 5
```

An maybe tweak these last `asyncio.sleep(5)` calls to ensure manual update subsequent calls are delayed according to the moment of manual calls.

Reply:
Very good tests :ok_hand: